### ISSUE TITLE
Publish new version for Bolts 1.9.0 Compatibility

### ISSUE THREAD

**Issue by felix-dumit** [parse-community/ParseFacebookUtils-iOS#89]
The Bolts `1.9.0` release removes the `NSException` related methods.
These are still present in the `1.11.1` (latest) version of this library and cause a compiler error if used with the new Bolts version.

This is already fixed in the master branch (https://github.com/parse-community/ParseFacebookUtils-iOS/commit/9a064914a25388f56b80576699aa04502a4604a3) so all that is needed is a new release.

**Comment by rgomesbr** [comment]
Waiting for this fix/new release as well.

**Comment by Steeve1995** [comment]
Yes please, would really appreciate it!

**Comment by rgomesbr** [comment]
Hello, @nlutsenko. Is there anything we can do to help you get the a new updated version released, using the fixes mentioned by @felix-dumit?
PS: I'm genuinely asking. I have never generated an update to anything on GitHub, let alone anything that's available on CocoaPods or else.

**Comment by flovilmart** [comment]
I’ll take care of new versions today

**Comment by rgomesbr** [comment]
@flovilmart , thanks for the new version. One issue though: when running `pod update` now, for CocoaPods, it downgrades the Facebook dependencies:

```
Installing FBSDKCoreKit 4.28.0 (was 4.29.0)
Installing FBSDKLoginKit 4.28.0 (was 4.29.0)
```

I think the issue here (https://github.com/CocoaPods/Specs/blob/master/Specs/8/b/6/ParseFacebookUtilsV4/1.11.2/ParseFacebookUtilsV4.podspec.json) is that the podspec should link the dependencies as:

```
"FBSDKLoginKit": [ "~> 4.28" ]
"FBSDKCoreKit": [ "~> 4.28" ]
```

instead of

```
"FBSDKLoginKit": [ "~> 4.28**.0**" ]
"FBSDKCoreKit": [ "~> 4.28**.0**" ]
```

**Comment by flovilmart** [comment]
There is an issue with 4.29 and the SDK would not build. If you’re using pods, I’m in the process to provide this as a subspec

**Comment by rgomesbr** [comment]
@flovilmart  Ok then!
Thanks for the explanation. 👍

**Comment by natanrolnik** [comment]
I think we can close this one now that the new version is out?

**Comment by flovilmart** [comment]
@natanrolnik yep! @rgomesbr we'll move the repo as well as the twitter utils in a single codebase: https://github.com/parse-community/Parse-SDK-iOS-OSX/pull/1230

This makes the CI and management easier for the future, this way, we'll also make sure every part is 'sane'.

**Comment by flovilmart** [comment]
@rgomesbr this is the error that I have with 4.29: - NOTE  | [tvOS] [Parse/FacebookUtils] xcodebuild:  FBSDKShareKit/FBSDKShareKit/FBSDKShareKit/FBSDKShareKit.h:25:9: fatal error: 'FBSDKShareKit/FBSDKShareMediaContent.h' file not found

**Comment by rgomesbr** [comment]
@flovilmart nice to know about moving it all to a single codebase :-)

Regarding the issue related to SDK 4.29, I see now what you meant... Anyway, I can keep using 4.28 for my iOS app while they don't fix this issue for 4.30 or else.

Once again, thanks for all the attention.

**Comment by flovilmart** [comment]
No problem, does it make sense to publish Facebook and twitter utils as subspecs? https://github.com/parse-community/Parse-SDK-iOS-OSX/blob/a69ef55b0a866886761ad6415b2306131b75d97b/README.md

**Comment by rgomesbr** [comment]
@flovilmart for me, imho, it does. It makes the `podfile` clearer, meaning:
I need Parse, so `pod 'Parse'`; since I also need Facebook, `pod 'Parse/FacebookUtils'`. Better than the way it goes now, in my podfile, using `ParseFacebookUtils-iOS`, where I do it like this (to keep _reminders_ of the dependencies for other developers on the team):

```
# No need to invoke Parse Framework here, since it's automatically loaded as dependency of ParseFacebookUtilsV4
# pod 'Parse', '~> 1.8'
pod 'ParseFacebookUtilsV4', '~> 1.11'

# No need to invoke Facebook SDK here, since they're automatically loaded as dependencies of ParseFacebookUtilsV4
# pod 'FBSDKCoreKit', '~> 4.11'
# pod 'FBSDKLoginKit', '~> 4.11'
```

I'll check it out later so that I can migrate my code to the 'new' subspec.
